### PR TITLE
cd: add Dockerfile and autobuild to ghcr.io

### LINKED ISSUE #62: add Dockerfile and autobuild to ghcr.io

### PR DESCRIPTION
**Pull request by LenaIkra**
This PR adds `Dockerfile` to our project and adds `build-new-image.yml` GitHub Actions Workflow file which implements new image build upon every push to `main`:
- cd: add initial Dockerfile
- cd: fix Dockerfile to rely on install.sh
- ci: add build-new-image.yml to automate new image builds
- [pre-commit.ci] auto fixes from pre-commit.com hooks
- ci: add meaningful tags in build-new-image.yml

Fixes #62

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,28 @@
+FROM ubuntu:latest
+
+LABEL org.opencontainers.image.source https://github.com/mlteamurfu2325/practicum-s1
```
Comment (khit-mle) [inline Dockerfile:3]: Apart from adding `LABEL` to link our image to repo, I think we should also add this from GHCR.io's docs:
```
LABEL org.opencontainers.image.description DESCRIPTION
```

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ENV DIR_NAME=practicum
+ENV GIT_BRANCH=main
+ENV LLM_SUMMARIZER=n
+ENV RUN_APP=n
+
+# Run the installation script with the provided arguments
+RUN ./install.sh --dir_name=$DIR_NAME --git_branch=$GIT_BRANCH --llm_summarizer=$LLM_SUMMARIZER --run_app=$RUN_APP
+
+# Set the default command to run when the container starts
+CMD ["/bin/bash", "-c", "source /root/practicum/.venv-practicum/bin/activate && cd /root/practicum/src && streamlit run run_app.py"]
```
Comment (khit-mle) [inline Dockerfile:28]: A bit hack-y, but I think it's OK for MVP ✅

**Review by khit-mle — APPROVED** [review]
## Code Review of [PR-55](https://github.com/mlteamurfu2325/practicum-s1/pull/55) by @khit-mle 

### Overview
- [x] The code follows the project's coding conventions and style guide.
- [x] The code is well-structured, readable, and maintainable.
- [x] The code is properly documented with clear comments and docstrings.
- [x] The code is efficient and optimized for performance, if applicable.
- [x] The code handles edge cases and potential errors gracefully.
- [x] The code is free of any security vulnerabilities or potential risks.

### Functionality
- [x] The code implements the required functionality as described in the PR description.
- [x] The code handles all expected inputs and outputs correctly.
- [x] The code integrates well with the existing codebase and does not introduce any conflicts.
- [x] The code does not introduce any new bugs or regressions.

### Testing
- [x] The code passes all existing tests and any newly added tests.
- [x] The tests cover a sufficient range of scenarios and edge cases (needed if new tests are added).
- [x] The tests are well-structured, readable, and maintainable (needed if new tests are added).

### Documentation
- [x] The code includes clear and concise comments explaining complex logic or algorithms.
- [x] The code includes docstrings for functions, classes, and modules, following the project's documentation standards.
- [x] The PR description provides a clear explanation of the changes made and their purpose.
- [x] Any necessary updates to the project's documentation (README, API docs, etc.) are included.

### Suggestions and Improvements
- [ ] Are there any areas where the code could be further optimized or improved?
- [ ] Are there any best practices or design patterns that could be applied to enhance the code?
- [ ] Are there any potential edge cases or error scenarios that have not been considered?
- [ ] Are there any additional tests that could be added to improve code coverage?

### Other Comments
@LenaIkra Thank you for your PR#55.

## Overall Recommendation:
- [x] Approve
- [ ] Request Changes
- [ ] Comment Only

**Review by alexa313 — APPROVED** [review]
## Code Review of [PR-55](https://github.com/mlteamurfu2325/practicum-s1/pull/55) by @alexa313

### Overview
- [x] The code follows the project's coding conventions and style guide.
- [x] The code is well-structured, readable, and maintainable.
- [x] The code is properly documented with clear comments and docstrings.
- [x] The code is efficient and optimized for performance, if applicable.
- [x] The code handles edge cases and potential errors gracefully.
- [x] The code is free of any security vulnerabilities or potential risks.

### Functionality
- [x] The code implements the required functionality as described in the PR description.
- [x] The code handles all expected inputs and outputs correctly.
- [x] The code integrates well with the existing codebase and does not introduce any conflicts.
- [x] The code does not introduce any new bugs or regressions.

### Testing
- [x] The code passes all existing tests and any newly added tests.
- [x] The tests cover a sufficient range of scenarios and edge cases (needed if new tests are added).
- [x] The tests are well-structured, readable, and maintainable (needed if new tests are added).

### Documentation
- [x] The code includes clear and concise comments explaining complex logic or algorithms.
- [x] The code includes docstrings for functions, classes, and modules, following the project's documentation standards.
- [x] The PR description provides a clear explanation of the changes made and their purpose.
- [x] Any necessary updates to the project's documentation (README, API docs, etc.) are included.

### Suggestions and Improvements
- [ ] Are there any areas where the code could be further optimized or improved?
- [ ] Are there any best practices or design patterns that could be applied to enhance the code?
- [ ] Are there any potential edge cases or error scenarios that have not been considered?
- [ ] Are there any additional tests that could be added to improve code coverage?

### Other Comments
@LenaIkra I agree with @khit-mle that your PR looks valid.

## Overall Recommendation:
- [x] Approve
- [ ] Request Changes
- [ ] Comment Only
